pkg/master/master.go

```diff
@@ -587,6 +607,15 @@ func (m *Master) removeAllThirdPartyResources(registry *thirdpartyresourcedataet
 	return nil
 }
 
+func (m *Master) removeAllThirdPartyResourceRoutes(ws *restful.WebService, kind string) {
+	routes := ws.Routes()
+	for _, route := range routes {
+		if strings.Contains(route.Path, strings.ToLower(kind)+"s") {
```

Comment:
I am not sure why do you need to do strings.ToLower(kind) + "s" here.
If I understand correctly, you are actually passing the resource name here to this function which should already have the "s" at end? You are extracting the resource name from the path itself (in getThirdPartyGroupKind)